var/spack/repos/builtin/packages/cbqn/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Cbqn(MakefilePackage):
+    """A BQN implementation in C"""
+
+    # Alternative homepage not specific to this implementation:
+    # https://mlochbaum.github.io/BQN/index.html
+    homepage = "https://github.com/dzaima/CBQN"
+    git = "https://github.com/dzaima/CBQN"
+
+    maintainers("ashermancinelli")
+
+    # Licenses listed in the order they appear in the repo's licensing section
+    # https://github.com/dzaima/CBQN/tree/master#licensing
+    license(
+        "MIT AND Apache-2.0 AND BSL-1.0 AND LGPL-3.0-only AND GPL-3.0-only AND MPL-2.0",
+        checked_by="ashermancinelli",
+    )
+
+    version("master", branch="master")
+    version("develop", branch="develop")
+    version("0.7.0", tag="v0.7.0")
+
+    depends_on("c", type="build")
+
+    variant("repl", default=True, description="Enable built-in REPL")
+    variant("ffi", default=True, description="Enable FFI support")
+    variant(
+        "build_type",
+        values=("o3n", "o3", "o3g", "c", "shared-o3", "debug", "static-bin", "static-lib"),
```

Comment:
What flags does `o3n` support? I suspect there isn't any difference between this and `o3` if you're using the target corrosponding to the machine, since Spack automatically sets the `-march` flags which imply the vectorization flags.

Reply:
I will ask the developer about it.

Reply:
The performance-sensitive parts of CBQN are written in [Singeli](https://github.com/mlochbaum/Singeli), which can't use `-march` as it doesn't have an architecture database (would be a pretty big maintainence burden). It reads CPU flags from /proc/cpuinfo. Passing -march to C but not allowing Singeli to get the native architecture would not be ideal, because performance would regress as features are implemented in Singeli and then forced to use the generic implementation.

Reply:
I understood this as a summary of the conversation to resolve it:

While not satisfactory, I think the question has been answered in a way that suggests that `o3n` is needed for good performance and no further changes appear to be needed.

Reply:
This is correct for v0.7.0. In the develop branch now and the next version when it comes out, we've worked out how to call the C compiler and parse its preprocessing headers for feature flags. Building with `target_from_cc=1` will get native performance using just the architecture passed in through C flags, not checking other sources.